src/appleseed/foundation/math/dual.h

```diff
@@ -50,7 +50,12 @@ class Dual
     typedef T ValueType;
 
     // Constructors.
-    Dual();
+    #if !defined(_MSC_VER) || _MSC_VER >= 1800
```

Comment:
That's debatable, but in the rest of the code we don't indent preprocessor statements and for consistency I would recommend doing the same here.

Reply:
A more important point: I actually hadn't touched this code on purpose. This changes the behavior as `m_has_derivates` is no longer set to `false` by the default constructor. Are you sure this is safe in all cases?

Reply:
Dual is inmutable. When you construct a Dual with a vector or with a vector with derivatives, m_has_derivatives is correctly set.

The change in the PR changes only the behaviour of default constructed Duals.
If we happen to use uninitialized Duals somewhere in the code that would be bug.